gap/examples.gi

```diff
@@ -370,3 +370,48 @@ GeneralisedPetersenGraphCons);
 
 InstallMethod(GeneralisedPetersenGraph, "for integer, integer", [IsInt, IsInt],
 {n, k} -> GeneralisedPetersenGraphCons(IsImmutableDigraph, n, k));
+
+InstallMethod(KingsGraphCons,
+"for IsMutableDigraph and two positive integers",
+[IsMutableDigraph, IsPosInt, IsPosInt],
+function(filt, n, k)
+  local D, i, j;
+  D := EmptyDigraph(filt, n * k);
+  for i in [1 .. k] do
+    for j in [1 .. (n - 1)] do
+      DigraphAddEdge(D, [((i - 1) * n) + j, ((i - 1) * n) + j + 1]);
+    od;
+  od;
+  for i in [1 .. (k - 1)] do
+    for j in [1 .. n] do
+      DigraphAddEdge(D, [((i - 1) * n) + j, ((i - 1) * n) + j + n]);
+    od;
+  od;
+  for i in [1 .. (k - 1)] do
+    for j in [1 .. (n - 1)] do
+      DigraphAddEdge(D, [((i - 1) * n) + j, ((i - 1) * n) + j + (n + 1)]);
+      DigraphAddEdge(D, [((i - 1) * n) + j + 1, ((i - 1) * n) + j + n]);
+    od;
+  od;
+  return DigraphSymmetricClosure(D);
```

Comment:
I think, once `TriangularGridGraph` is merged, this function should proceed as follows:
```suggestion
  D := TriangularGridGraph(IsMutableDigraph, n, k);
  for i in [1 .. (k - 1)] do
    for j in [1 .. (n - 1)] do
      a := ((i - 1) * n) + j;
      b := ((i - 1) * n) + j + (n + 1);
      DigraphAddEdge(D, a, b);
      DigraphAddEdge(D, b, a);       
    od;
  od;
  return D;
```
This will also mean that the digraph vertex labels are properly set, to give the correspondence between vertices and grid positions.

Reply:
Right, that makes sense! Another consideration is that I'll be working on graph products next, and there is an even simpler way to construct a king's graph using the strong product of two path graphs. So I'll make a note to update the Kingsgraph function once the strong product function exists.

Reply:
Oh that's brilliant!